flink-runtime/src/test/java/org/apache/flink/runtime/state/OperatorStateBackendTest.java

```diff
@@ -355,10 +355,11 @@ public void testSnapshotEmpty() throws Exception {
 		CheckpointStreamFactory streamFactory =
 				abstractStateBackend.createStreamFactory(new JobID(), "testOperator");
 
-		RunnableFuture<OperatorStateHandle> snapshot =
+		RunnableFuture<SnapshotResult<OperatorStateHandle>> snapshot =
 				operatorStateBackend.snapshot(0L, 0L, streamFactory, CheckpointOptions.forCheckpoint());
 
-		OperatorStateHandle stateHandle = FutureUtil.runIfNotDoneAndGet(snapshot);
+		SnapshotResult<OperatorStateHandle> snapshotResult = FutureUtil.runIfNotDoneAndGet(snapshot);
+		OperatorStateHandle stateHandle = snapshotResult != null ? snapshotResult.getJobManagerOwnedSnapshot() : null;
```

Comment:
This seems a bit circumstantial. Can't we make it that `snapshotResult` is always not null?

Reply:
👍